### DISCUSSION TITLE
Packet error rate in Data rate channel seems to have no effect

### DISCUSSION THREAD

**Discussion by WolframStrauss** [inet-framework/inet#909] [General]
Hi, I try to use `DatarateChannel` to model a link packet error rate (used file see below). The PER setting seems to have no effect , which I take from the number of created and received packets at the producer and collector block.

What do I miss here, or is there another simple module (I just need a PER of x %), which I can use?

Thank you. Kind regards. Wolfram

```ned code
import inet.queueing.sink.PassivePacketSink;
import inet.queueing.source.ActivePacketSource;
import ned.DatarateChannel;

network  OnDatarateChannel {
    submodules:
        producer: ActivePacketSource {
            @display("p=100,100");
        }
        collector: PassivePacketSink {
            @display("p=300,100");
        }
    connections :
        // Note: per=0.5; seems to have no effect
        producer.out --> DatarateChannel {datarate = 160bps; delay = 0ms; per=0.5;} --> collector.in;
}
```

```ini code
[Config OnDatarateChannel]
network = OnDatarateChannel
sim-time-limit = 10s

# Channel parameters see: `OnDatarateChannel.ned` file
*.producer.productionInterval = 1s
*.producer.packetLength = 1B
```

**Comment by avarga** [comment]
The packet error rate parameter of DatarateChannel works, as evidenced by `test/core/cDatarateChannel_per_1.test`, but it doesn't cause packets to be dropped on link. It just sets a "has bit error" flag in them. You can read that flag from C++ code using `pkt->hasBitError()`.

**Reply by WolframStrauss** [reply]
Well, it is a packet error rate, not a packet drop rate. My head was taking a wrong turn somewhere. Thank you.
